### PR TITLE
don't abort on batch write failures if flag is not specified for copy command

### PR DESCRIPTION
**Pull request by jteh**
this addresses issue #628

### CODE REVIEW

File: flux-cli/src/test/java/com/marklogic/flux/impl/copy/CopyTest.java [hunk 1]
```diff
@@ -100,6 +101,44 @@ void sameDatabaseWithMetadata() {
         assertDirectoryCount("/copied/", 15);
     }
 
+    @Test
+    void runsToCompletionWithBatchWriteFailuresAndAbortOnWriteFailureNotSpecified() {
```
Comment (rjrudin) [inline flux-cli/src/test/java/com/marklogic/flux/impl/copy/CopyTest.java:105]: Add an assertion to verify that the collection size is zero - e.g. `assertCollectionSize("author-copies", 0)`. 

We also want an assertion here to make sure that stderr doesn't contain an error. You can use `runAndReturnStderr` for that, and then assertFalse that stderr contains "Role does not exist".

File: flux-cli/src/test/java/com/marklogic/flux/impl/copy/CopyTest.java [hunk 2]
```diff
@@ -100,6 +101,44 @@ void sameDatabaseWithMetadata() {
...
+        run(CommandLine.ExitCode.OK,
+            "copy",
+            "--collections", "author",
+            "--partitions-per-forest", "1",
+            "--categories", "content,metadata",
+            "--connection-string", makeConnectionString(),
+            "--output-connection-string", makeConnectionString(),
+            // No need to specify permissions since they are included via "--categories".
```
Comment (rjrudin) [inline flux-cli/src/test/java/com/marklogic/flux/impl/copy/CopyTest.java:113]: Remove this comment as it's not correct.

File: flux-cli/src/test/java/com/marklogic/flux/impl/copy/CopyTest.java [hunk 3]
```diff
@@ -100,6 +101,44 @@ void sameDatabaseWithMetadata() {
...
+        assertStderrContains(
+            "Role does not exist: sec:role-name = invalid-roleZZZ",
+            CommandLine.ExitCode.SOFTWARE,
+            "copy",
+            "--collections", "author",
+            "--partitions-per-forest", "1",
+            "--categories", "content,metadata",
+            "--connection-string", makeConnectionString(),
+            "--output-connection-string", makeConnectionString(),
+            // No need to specify permissions since they are included via "--categories".
```
Comment (rjrudin) [inline flux-cli/src/test/java/com/marklogic/flux/impl/copy/CopyTest.java:133]: Remove this comment as well.

**Review by rjrudin — CHANGES_REQUESTED** [review]
@jteh Thanks for catching this and for the PR! Just a few small comments, and also change the PR to point to the `develop` branch instead of `main`. This fix will be included in our next patch or minor release.

File: flux-cli/src/test/java/com/marklogic/flux/impl/copy/CopyTest.java (hunk 1 above)
Comment (jteh) [reply flux-cli/src/test/java/com/marklogic/flux/impl/copy/CopyTest.java:105]: Thanks @rjrudin , I've addressed comments and also improved this test by specifying a different db to copy to and ensuring that db is empty prior to test execution. This ensures the assertion works as expected as the db is clean.

I'm also closing this PR to be superceded by #630 so that the changes target the develop branch.

**Comment by jteh** [comment]
superceded by #630
